### ISSUE TITLE
 SRTP documentation

### ISSUE THREAD

**Issue by hmoghani** [SIPp/sipp#536]
Hi @jeannotlanglois ,

I was wondering if you could provide the documentation for the SRTP feature. I saw "[Detailed feature documentation available upon request]" but I couldn't find a another way to contact you. I'd really appreciate it if you could provide any documents you have regarding the SRTP feature. 

Thanks.

**Comment by jeannotlanglois** [comment]
Hello:

Sorry for not being able to answer your email earlier - the pandemic makes me extremely overloaded in an understaffed team.  But I finally got to it today.

I expect SIPP-RTPCHECK RTP/SRTP gateway bit pattern testing functionality to be merged into the next SIPP version (3.7).  There is one more pull request (#538, still unmerged as of 2021/09/06) which I have rebased today.  In that pull request a full PDF is provided which describes in great detail the details of RTP/SRTP gateway bit pattern testing functionality:

https://github.com/jeannotlanglois/sipp/blob/sipp-36x-rtpcheck/rtpcheck_xml_syntax_reference.pdf


(Obviously once this pull request #538 is merged into the main SIPP-3.7 branch this PDF and a few important bugfixes should be visible there.)

Cheers :)

--
Jeannot Langlois
Software Developer
SIP Inspector
RFC Lawyer
Refactoring Expert
MiVoice Border Gateway Development
Mitel Networks
4000 Innovation Drive
Kanata (Ontario)
K2K 3K1
http://www.mitel.com<http://www.mitel.com/>
***@***.******@***.***>
x444364
(613) 691-3385 [Direct Dial]

"Never give in, never give in, never, never, never, never.  In nothing, great or small, large or petty.  Never give in except to convictions of honor and good sense."
- Winston Churchill

"Can I get an update on issue X?  Did you fix issue X in stream Y?"
[https://mitel.atlassian.net/issues/?filter=17325]

"When is MBG version X planned for GA?  Will it include fix Y?"
[https://mitel.atlassian.net/wiki/spaces/MBG/pages/5588287492/Project+Dashboard]

From: hmoghani ***@***.***>
Sent: Wednesday, August 18, 2021 5:19 PM
To: SIPp/sipp ***@***.***>
Cc: Jeannot Langlois ***@***.***>; Mention ***@***.***>
Subject: [SIPp/sipp] SRTP documentation (#536)


Hi @jeannotlanglois<https://can01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fjeannotlanglois&data=04%7C01%7Cjeannot.langlois%40mitel.com%7Ce57427f604d340213dd808d9628dd732%7C4bff5a2bb30d493981ff8f76138347df%7C1%7C0%7C637649183605090759%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=lmTsx%2BYNhQBKb7GPXtUK4vU8Vks17%2FACZtWxgpDUlaA%3D&reserved=0> ,

I was wondering if you could provide the documentation for the SRTP feature. I saw "[Detailed feature documentation available upon request]" but I couldn't find a another way to contact you. I'd really appreciate it if you could provide any documents you have regarding the SRTP feature.

Thanks.

-
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://can01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FSIPp%2Fsipp%2Fissues%2F536&data=04%7C01%7Cjeannot.langlois%40mitel.com%7Ce57427f604d340213dd808d9628dd732%7C4bff5a2bb30d493981ff8f76138347df%7C1%7C0%7C637649183605100757%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=CEasmLxsh%2BG6wj%2B1RjooryalkEJUkiMnkxV%2FrUfPUWo%3D&reserved=0>, or unsubscribe<https://can01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAJK3K5LKNUY7PHKMKNEE5WDT5QPVNANCNFSM5CM32MAQ&data=04%7C01%7Cjeannot.langlois%40mitel.com%7Ce57427f604d340213dd808d9628dd732%7C4bff5a2bb30d493981ff8f76138347df%7C1%7C0%7C637649183605110749%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=8gCosQn2B9wTgx%2BQVJS8VDjB2HcxPaIIq118sCdAKKk%3D&reserved=0>.
Triage notifications on the go with GitHub Mobile for iOS<https://can01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fapps.apple.com%2Fapp%2Fapple-store%2Fid1477376905%3Fct%3Dnotification-email%26mt%3D8%26pt%3D524675&data=04%7C01%7Cjeannot.langlois%40mitel.com%7Ce57427f604d340213dd808d9628dd732%7C4bff5a2bb30d493981ff8f76138347df%7C1%7C0%7C637649183605110749%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=WvvuBLXSEbh3izxzFtIk2riKYdduhiIrhNTvhXMZVwM%3D&reserved=0> or Android<https://can01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fplay.google.com%2Fstore%2Fapps%2Fdetails%3Fid%3Dcom.github.android%26utm_campaign%3Dnotification-email&data=04%7C01%7Cjeannot.langlois%40mitel.com%7Ce57427f604d340213dd808d9628dd732%7C4bff5a2bb30d493981ff8f76138347df%7C1%7C0%7C637649183605120742%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=wBJrMKXtS9804DxU5nF4p8AOW8ljAy1p5dFWtezKf0g%3D&reserved=0>.

________________________________
NOTE: This e-mail (including any attachments) is for the sole use of the intended recipient(s) and may contain information that is confidential and/or protected by legal privilege. Any unauthorized review, use, copy, disclosure or distribution of this e-mail is strictly prohibited. If you are not the intended recipient, please notify Mitel immediately and destroy all copies of this e-mail. Mitel does not accept any liability for breach of security, error or virus that may result from the transmission of this message.

**Comment by hmoghani** [comment]
Hi,

I really appreciate your response. Thank you. I look forward to having the
new feature in version 3.7.
And also thanks for your hard work for the sipp community.

Hamid

**Comment by jeannotlanglois** [comment]
No problem.  Working for MITEL's B2BUA Border Gateway required to troubleshoot countless issues related to SRTP and since there are so many permutations of how it can be done in the SDP got me to get busy adding that support to SIPP between 2015-2018 so that it could simplify my job.

My hopes are that having ways to test it with SIPP helps the greater good of the community for interoperability with SRTP since it is growing in popularity/need and a lot of telecoms are moving to SIP.

At this stage it only supports AES_128_CM_SHA1_80 and AES_128_CM_SHA1_32 but it is quite a huge step forward to help with testing all kinds of various SIP endpoints.

I'm not fully satisfied with the code yet so I eventually hope I can modularize/refactor it in an even more maintainable piece of code that is also better documented (in the code).

Eventually hopefully find some time to support other stronger crypto suites.

Sadly I will likely not be able to have enough time to answer support questions about it very often but I hope the documentation covers most needs.

I'm not sure when the SIPP-3.7 release is planned (I think this would be a question for Walter Doekes), but as far as my understanding goes it is the next one.

Enjoy :)

--
Jeannot Langlois
Software Developer
SIP Inspector
RFC Lawyer
Refactoring Expert
MiVoice Border Gateway Development
Mitel Networks
4000 Innovation Drive
Kanata (Ontario)
K2K 3K1
http://www.mitel.com<http://www.mitel.com/>
***@***.******@***.***>
x444364
(613) 691-3385 [Direct Dial]

"Never give in, never give in, never, never, never, never.  In nothing, great or small, large or petty.  Never give in except to convictions of honor and good sense."
- Winston Churchill

"Can I get an update on issue X?  Did you fix issue X in stream Y?"
[https://mitel.atlassian.net/issues/?filter=17325]

"When is MBG version X planned for GA?  Will it include fix Y?"
[https://mitel.atlassian.net/wiki/spaces/MBG/pages/5588287492/Project+Dashboard]
